paperpi/library/Plugin.ipynb

```diff
@@ -141,7 +144,131 @@
     "        self.max_priority = max_priority\n",
     "        self.hash = self._generate_hash()\n",
     "        \n",
-    "    \n",
+    "        @property\n",
+    "        def resolution(self):\n",
+    "            return self._resolution\n",
+    "\n",
+    "        @resolution.setter\n",
+    "        def resolution(self, value):\n",
+    "            if not isinstance(value, tuple) or len(value) != 2 or not all(isinstance(x, int) and x > 0 for x in value):\n",
+    "                raise TypeError(\n",
+    "                    \"resolution should be a tuple of positive integers.\")\n",
+    "            self._resolution = value\n",
+    "\n",
+    "        @property\n",
+    "        def name(self):\n",
+    "            return self._name\n",
+    "\n",
+    "        @name.setter\n",
+    "        def name(self, value):\n",
+    "            if value is not None and not isinstance(value, str):\n",
+    "                raise TypeError(\"name should be a string or None.\")\n",
+    "            self._name = value\n",
+    "\n",
+    "        @property\n",
+    "        def layout(self):\n",
+    "            return self._layout\n",
+    "\n",
+    "        @layout.setter\n",
+    "        def layout(self, value):\n",
+    "            if not isinstance(value, dict):\n",
+    "                raise TypeError(\"layout should be a dictionary.\")\n",
+    "            self._layout = value\n",
+    "\n",
+    "        @property\n",
+    "        def update_function(self):\n",
+    "            return self._update_function\n",
+    "\n",
+    "        @update_function.setter\n",
+    "        def update_function(self, value):\n",
+    "            if value is not None and not callable(value):\n",
+    "                raise TypeError(\n",
+    "                    \"update_function should be a function or None.\")\n",
+    "            self._update_function = value\n",
```

Comment:
this duplicates the setter on line 249; I don't think this was intended.

Reply:
Okay I will remove these lines